docs/DeveloperGuide.md

```diff
@@ -1,38 +1,360 @@
+4) If the user wants to update Modular Credits, the `editModularCredits()` method is called.
+   This method directly modifies the `modularCredits` attribute of the `Module` object.
+5) If the user wants to update Module Type, the `editModuleType()` method is called.
+   This method removes the existing `Module` object and adds a new `Module` object of the new type specified
+   by the user. The `getModuleCode()`, `getModularCredits()`, `getYear()`, `getSemester()` and `getGrade()`
+   methods in the `Module` class are also called upon to obtain the respective fields required to create the new
+   `Module` object.
+6) If the user wants to update the Year, the `editYear()` method is called.
+   This method directly modifies the `year` attribute of the `Module` object.
+7) If the user wants to update the Semester, the `editSemester()` method is called.
+   This method directly modifies the `semester` attribute of the `Module` object.
+
+![editCommand](diagrams/EditCommand.png)
+
+<small><i>Figure ???</i></small>
+
+### 3.7. Add or Update Grade
+
+`grade` command:
+
+The `grade` command is used by the user to update the grade of modules that have already been completed.
+
+The sequence by which the Parser class handles the `grade` command is as follows:
+1) The `Parser` class extracts the other fields of the user input, and calls upon the `updateModuleGrade()` method
+   in the `ModuleList` class.
+2) The method then loops through the moduleList array to find the moduleCode that requires a grade update.
+3) If the module is found, it calls the `Module` object method `setGrade()` to update the grade field
+   and prints a success message by calling the `printUpdatedModuleGrade()` method of the `Print` class.
+4) Else, it prints an error message by calling the `printInvalidModule()` method of the `Print` class.
+
+The sequence of events above can be represented with the following sequence diagram:
+
+![gradeCommand](diagrams/GradeCommand.png)
+
+<small><i>Figure ???</i></small>
+
+
+### 3.8. Calculate CAP
+
+`calculateCAP` command:
+
+The `calculateCAP` command is used by the user to calculate the Cumulative Average Point (CAP) of graded modules.
+
+The sequence by which the Parser class handles the `calculateCAP` command is as follows:
+1) The `Parser` class extracts the other fields of the user input, and calls upon the `calculateCAP()` method
+   in the `ModuleList` class.
+2) The method then loops through the moduleList array to find modules that have been graded.
+3) It checks if the module grade should be counted by self-invocation of the `shouldCountModule()` method
+4) If it is, proceeds to obtain the grade value by self-invocation of the `getGradeValue()` method.
+5) Calculates the required grade values, and finally calls the `printCalculatedCAP()` of the `Print` class to display
+   the result to the user
+
+The sequence of events above can be represented with the following sequence diagram:
+
+![calculateCAP](diagrams/CalculateCAP.png)
```

Comment:
Good that self invocation bars are included. Should considering including them for the other diagrams too
<img width="245" alt="image" src="https://user-images.githubusercontent.com/88694646/228723677-000b61b4-9ab0-48ec-9f0f-acd9e37bd2c8.png">